src/libmongoc/src/mongoc/mongoc-cmd.c

```diff
@@ -785,6 +785,29 @@ _is_retryable_read (const mongoc_cmd_parts_t *parts,
 }
 
 
+static bool
+_txn_in_progress (mongoc_cmd_parts_t *parts)
+{
+   mongoc_client_session_t *cs;
+
+   cs = parts->prohibit_lsid ? NULL : parts->assembled.session;
+   if (!cs) {
+      return false;
+   }
+
+   if (_mongoc_client_session_txn_in_progress (cs)) {
+      return true;
+   }
+
+   /* commitTransaction counts, too. */
+   if (parts->assembled.is_txn_finish) {
+      return true;
+   }
+
+   return false;
```

Comment:
I feel a ternary return helps understand the logic better after the early return above:

```suggestion
   cs = parts->prohibit_lsid ? NULL : parts->assembled.session;
   if (!cs) {
      return false;
   }

   return _mongoc_client_session_txn_in_progress (cs)
      /* commitTransaction counts as in progress, too. */
      || parts->assembled.is_txn_finish);
```

Reply:
Eh, I usually like to keep multi-step logic in distinct steps for readability, but I don't feel strongly about it in this situation and I am happy to use your suggestion instead.